tls/s2n_client_hello.c

```diff
@@ -270,6 +269,16 @@ int s2n_process_client_hello(struct s2n_connection *conn)
         S2N_ERROR(S2N_ERR_PROTOCOL_VERSION_UNSUPPORTED);
     }
 
+    if (conn->config->quic_enabled) {
+        ENSURE_POSIX(conn->actual_protocol_version >= S2N_TLS13, S2N_ERR_PROTOCOL_VERSION_UNSUPPORTED);
+
+        /* In TLS1.3, legacy_session_id is only set to indicate middlebox compatability mode.
+         * When running with QUIC, S2N does not support middlebox compatability mode.
+         * https://tools.ietf.org/html/draft-ietf-quic-tls-32#section-8.4
+         */
+        ENSURE_POSIX(conn->session_id_len == 0, S2N_ERR_BAD_MESSAGE);
```

Comment:
Not S2N_ERR_UNSUPPORTED_WITH_QUIC?

Reply:
See https://github.com/awslabs/s2n/pull/2373#discussion_r521646111